src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -411,129 +409,111 @@ export class NeynarAPIClient {
+      cursor = response.data.result.next.cursor;
     }
   }
 
   /**
-   * Fetch all likes by a given user.
+   * Fetch all likes by a given user. See [Neynar documentation](https://docs.neynar.com/reference/get-user-cast-likes)
    */
   public async *fetchUserCastLikes(
-    user: { fid: number },
+    fid: number,
     { pageSize = 100 } = {}
-  ): AsyncGenerator<ReactionWithCastMeta, void, undefined> {
+  ): AsyncGenerator<ReactionWithCastMeta[], void, undefined> {
     let cursor: string | undefined;
     let viewer: number | undefined;
 
     while (true) {
       // fetch one page of likes
       const response = await this.apis.user.userCastLikes(
-        user.fid,
+        fid,
         viewer,
         pageSize,
         cursor
       );
 
-      // Extract the likes array from the response (or provide an empty array as a default)
-      const likes =
-        response.data.result?.likes != null ? response.data.result.likes : [];
-
-      // Create an async iterable for the likes
-      for (const like of likes) {
-        yield like;
-      }
+      yield response.data.result.likes;
 
       // prep for next page
-      if (response.data.result?.next?.cursor === undefined) {
+      if (response.data.result.next?.cursor === null) {
         break;
       }
-      cursor = response.data.result.next.cursor ?? undefined;
+      cursor = response.data.result.next.cursor;
     }
   }
 
   /**
-   * Gets the specified user via their FID (if found)
+   * Gets the specified user via their FID (if found). See [Neynar documentation](https://docs.neynar.com/reference/get-user-information-by-fid)
    */
-  public async lookupUserByFid(fid: number): Promise<User | undefined> {
+  public async lookupUserByFid(fid: number): Promise<User | null> {
     try {
       const response = await this.apis.user.user(fid);
       return response.data.result?.user;
     } catch (error) {
       if (NeynarAPIClient.isApiErrorResponse(error)) {
-        if (error.response.status === 404) return undefined;
+        if (error.response.status === 404) return null;
       }
       throw error;
     }
   }
 
   /**
-   * Gets the specified user via their username (if found)
+   * Gets the specified user via their username (if found). See [Neynar documentation](https://docs.neynar.com/reference/get-user-information-by-username)
    */
-  public async lookupUserByUsername(
-    username: string
-  ): Promise<User | undefined> {
+  public async lookupUserByUsername(username: string): Promise<User | null> {
     let viewer: number | undefined;
-    const response = await this.apis.user.userByUsername(username, viewer, {
-      validateStatus: (status: number) => {
-        return status === 200 || status === 404;
-      },
-    });
-    if (response.status === 404) return undefined;
-    return response.data.result?.user;
+    const response = await this.apis.user.userByUsername(username, viewer);
+    return response.data.result.user ?? null;
```

Comment:
Let's add this to `eslint.rc`, which should catch all errors of this category going forward:
```json
"rules": {
    "@typescript-eslint/no-unnecessary-condition": ["error", {
      "allowConstantLoopConditions": true
    }]
```

Reply:
agreed, this works except for in `lookupUserByUsername` which returns a result {} if not found, so result.user is undefined

Reply:
The spec says `user` cannot be undefined: https://github.com/standard-crypto/farcaster-js/blob/9cf0eb61da20fed090e127364442e822d6630f15/src/neynarAPI/neynarV1API/openapi/spec.yaml#L668-L674

Reply:
It does, but apparently that doesn't matter:

```
  NeynarAPI
    #lookupUserByUsername
logging lookupUserByUsername response.data: {"result":{}}
logging lookupUserByUsername response.data.result.user: undefined
      1) returns null if user not found


  0 passing (482ms)
  1 failing

  1) NeynarAPI
       #lookupUserByUsername
         returns null if user not found:
     AssertionError: expected undefined to be null
```

Reply:
Noting we're going to run this by the Neynar team -- at the very least, the spec is incorrect. It's possible they also want to change the server to return `null` instead of `undefined`

Reply:
Remind me what was the conclusion here